src/components/modals/NumericInput.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react'
+import { useRef } from 'react'
+import { TextInput, TextInputProps } from 'react-native'
+
+import { useHandler } from '../../hooks/useHandler'
+import { formatNumberInput, formatToNativeNumber, isValidInput } from '../../locales/intl'
+import { useState } from '../../types/reactHooks'
+
+type Props = {
+  ref: React.RefObject<TextInput>
+  minDecimals?: number
+  maxDecimals?: number
+} & TextInputProps
+
+export const NumericInput = React.forwardRef<TextInput, Props>((props: Props, ref) => {
+  const { onChangeText, minDecimals, maxDecimals, value, ref: _, ...rest } = props
+  const [innerValue, setInnerValue] = useState<string>(props.value ?? '')
+  const propValue = useRef(props.value ?? '')
+
+  const onChangeTextInner = useHandler(text => {
+    if (isValidInput(text)) {
+      const nativeNum = text === '' ? '' : formatToNativeNumber(text)
+      const displayNum = text === '' ? '' : formatNumberInput(nativeNum, { minDecimals, maxDecimals })
+      if (displayNum !== innerValue) {
+        if (onChangeText != null) onChangeText(nativeNum)
+        setInnerValue(displayNum)
+      }
+    }
+  })
+
+  if (props.value !== propValue.current) {
+    propValue.current = props.value ?? ''
+    const displayNum = props.value === '' ? '' : formatNumberInput(propValue.current, { minDecimals, maxDecimals })
+    setInnerValue(displayNum)
```

Comment:
I was going to approve the PR, but then I realized this line will crash React. You absolutely, 100% cannot call `setState` in the main render body. This logic must be wrapped inside `useEffect` or such, or React will throw an error.

Reply:
Hmm. Do you have documentation on this limitation. I definitely hit that code and I don’t get a crash or even a warning.